src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -97,9 +97,15 @@ public function process(ContainerBuilder $container)
 
             // We must assume that the class value has been correctly filled, even if the service is created by a factory
             $class = $container->getParameterBag()->resolveValue($def->getClass());
-
             $interface = 'Symfony\Component\EventDispatcher\EventSubscriberInterface';
+
             if (!is_subclass_of($class, $interface)) {
+                if (!class_exists($class, true)) {
+                    throw new \RuntimeException(
+                        sprintf('Class "%s" from definition of service "%s" can not be found. Check existence of file and correct naming', $class, $id)
```

Comment:
@nicolas-grekas , i don't understand :( Two most common class loading issues are:

1. You wrote a classname incorrect while filename is correct
1. You wrote a filename incorrect while classname is correct 

In first case you will see `RuntimeException` and explained message with most typical errors. If you will try to instantiate a class without a mapped file — fatal error will be raised, so `RuntimeException`  is much closer than `InvalidArgumentException`.